binlog_streamer.go

```diff
@@ -241,26 +269,36 @@ func (s *BinlogStreamer) FlushAndStop() {
 	// passed the initial target position.
 	err := WithRetries(100, 600*time.Millisecond, s.logger, "read current binlog position", func() error {
 		var err error
-		s.targetBinlogPosition, err = ShowMasterStatusBinlogPosition(s.DB)
+		s.stopAtBinlogPosition, err = ShowMasterStatusBinlogPosition(s.DB)
 		return err
 	})
 
 	if err != nil {
 		s.ErrorHandler.Fatal("binlog_streamer", err)
 	}
-	s.logger.WithField("target_position", s.targetBinlogPosition).Info("current stop binlog position was recorded")
+	s.logger.WithField("target_position", s.stopAtBinlogPosition).Info("current stop binlog position was recorded")
 
 	s.stopRequested = true
 }
 
-func (s *BinlogStreamer) updateLastStreamedPosAndTime(ev *replication.BinlogEvent) {
-	if ev.Header.LogPos == 0 || ev.Header.Timestamp == 0 {
-		// This shouldn't happen, as the cases where it does happen are excluded.
-		// However, I've not seen all the cases yet.
-		s.logger.Panicf("logpos: %d %d %T", ev.Header.LogPos, ev.Header.Timestamp, ev.Event)
+func (s *BinlogStreamer) updateLastStreamedPosAndTime(ev *replication.BinlogEvent, pos mysql.Position, isResumablePosition bool) {
```

Comment:
It's a bit confusing to see this function is receiving an event (that has a position in the header), but also a separate position along with it. I think we can simplify this by dropping either the `pos` or `ev` (pick one):

```suggestion
func (s *BinlogStreamer) updateLastStreamedPosAndTime(pos mysql.Position, isResumablePosition bool) {
```
and then we can consistently just refer to a the position as opposed to looking at both. Alternatively, just extracting the `Timestamp` from the `replication.BinlogEvent` as an `evTime` or something to make it clearer that's all this method is using.

Another thought is that one param seems to be used for actually updating the last streamed and resumable position, and the other for instrumenting binlog streamer lag? Worth breaking these into separate methods?

Reply:
Good idea to extract and pass the timestamp.. Not gonna go down the instrumentation refactor for now if that's okay